src/compiler/checker.ts

```diff
@@ -2173,6 +2174,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         [".json", ".json"],
     ];
 
+    const symbolToStringWorkerSourceFilePrinter = pooled(() => createPrinter({ removeComments: true, neverAsciiEscape: true }), (printer) => printer.reset());
+    const symbolToStringWorkerNonSourceFilePrinter = pooled(() => createPrinter({ removeComments: true }), (printer) => printer.reset());

```

Comment:
It turns out that this is way overkill; the printer already internally resets itself after each public API call. So, this could just a plain variable (or a memoize call to make it lazy).